content/tutorials/WebRTC-file-transfers-with-Ably.textile

```diff
@@ -0,0 +1,509 @@
+        if (file) {
+            var mbSize = file.size / (1024 * 1024);
+            if (mbSize > MAX_FSIZE) {
+                alert("Your file is too big, sorry.");
+                // Reset file input
+                return;
+            }
+            fileReader.onload = (e) => { onReadAsDataURL(e, null, file) }
+            fileReader.readAsDataURL(file)
+            render()
+        }
+    }
+```
+"See this step in Github":https://github.com/samuelayo/tutorials/commit/23b5939f13f4e3249b73f074bdfa2643a573259e
+
+**What is going on in the code block above?**
+To share files with a user/member, we need to click the @chat now@ button next to their name. This button, in turn, calls the @prepareChat@ method, which we have just defined.
+
+The @prepareChat@ method receives a single argument, i.e, the @client-id@, which is the client you want to send files to. We then check to be sure that the user is not trying to send files to himself. Although it isn’t possible for the user to do this, we do this as an extra layer of security. At this point, if the user does not have an active connection with the other peer, it means he is the one initiating the chat/file transfer, so we create a new connection with that @client_id@, passing the initiator status to be true. After this, we call the @render@ method to render the files that might have been sent by the other peer.
+
+For the two pairs to connect, the other peer has to be signaled of the incoming connection. This is where our subscription comes in. We subscribe to an event specifically for the current client’s id, which we defined the event as @ablywebrtc-signal/${clientId}@. Here, we check if the current user has a connection instance of the current client trying to connect, if not, we initiate a new connection for the other party, and surely the initiator flag is set to false.  We then call on the @handleSignal@ method of the connection class, to handle the signal coming in and connect the two pairs.
+
+**Now, we can go ahead to send files.**
+
+In our @sendMessage@ method, we use the @currentChat@ variable to figure out who it is that we are trying to send the file to. We also check that the file about to be sent is not larger than our file limit. Here, it is set as 2mb, feel free to increase yours. We then initiate a new @fileReader@ instance. Because the @fileReader@ is an asynchronous API, we have to wait till it’s loaded before we can do anything with it. Here, we defined a new function for it to fire when it’s @onload@ event is ready. We passed the event, a null and the file we want to send to a function called @onReadAsDataURL@. We then ask the @fileReader@ to read the file as a DataUrl.
+
+
+p(tip). Note: Files can be read as @ArrayBuffers@, @DataUrls@, @BinaryStrings@, and @Text@. The same concept of sending the file in chunks follows for the other read types. In this lesson, we are using the DataUrl type. 
+
+**What happens in the onReadAsDataURL method?**
```

Comment:
<img width="812" alt="screen shot 2018-08-02 at 17 11 08" src="https://user-images.githubusercontent.com/5900152/43582653-be3c1e6e-967a-11e8-8e2e-d47ff73f428c.png">
The tip is overlapping the code when rendered.

Reply:
For this, i'm not sure of the fix. tried using multiple lines, comes up the same way. if you could help me with the way to solve this, i'll be glad